plugins/modules/zammad_ticket.py

```diff
@@ -0,0 +1,318 @@
+    required: true
+    type: str
+  title:
+    description: The title of the ticket.
+    required: true
+    type: str
+  group:
+    description: The group handling the ticket.
+    required: true
+    type: str
+  subject:
+    description: The subject for the ticket's article.
+    required: true
+    type: str
+  body:
+    description: The body content for the ticket's article.
+    required: true
+    type: str
+  internal:
+    description: Indicates whether the article is internal. Defaults to false.
+    required: false
+    type: bool
+    default: false
+  ticket_state:
+    description: The state of the ticket (e.g., open, pending, etc.).
+    required: true
+    type: str
+  priority:
+    description: The priority of the ticket (e.g., low, normal, high).
+    required: true
+	type: str
+
+examples:
+  - name: Create a new ticket
+    zammad_ticket:
+      fqdn: "https://zammad.example.com"
+      endpoint: "/api/v1/tickets"
+      api_user: "api_user"
+      api_secret: "api_secret"
+      state: "present"
+      title: "Internet Outage"
+      group: "Support"
+      customer: "customer@example.com"
+      subject: "Internet is down"
+      body: "The internet is not working since this morning."
+      internal: false
+      ticket_state: "open"
+      priority: "high"
+
+  - name: Update an existing ticket
+    zammad_ticket:
+      fqdn: "https://zammad.example.com"
+      endpoint: "/api/v1/tickets"
+      api_user: "api_user"
+      api_secret: "api_secret"
+      state: "present"
+      ticket_id: "12345"
+      title: "Internet Outage - Follow Up"
+      group: "Support"
+      customer: "customer@example.com"
+      subject: "Update on internet issue"
+      body: "The internet issue is being worked on."
+      internal: true
+      ticket_state: "pending"
+      priority: "normal"
+
+  - name: Close a ticket
+    zammad_ticket:
+      fqdn: "https://zammad.example.com"
+      endpoint: "/api/v1/tickets"
+      api_user: "api_user"
+      api_secret: "api_secret"
+      state: "absent"
+      ticket_id: "12345"
+
+return:
+  ticket_id:
+    description: The ID of the created or updated support ticket.
+    type: str
+    returned: always
+    sample: "12345"
+  status_code:
+    description: The status code returned by the Zammad API.
+    type: int
+    returned: always
+    sample: 200
+  message:
+    description: A message indicating the result of the operation (success or failure).
+    type: str
+    returned: always
+    sample: "Ticket created successfully."
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+import json
+import requests
+
+def make_request(method, fqdn, endpoint, api_user, api_secret, data, ticket_id = ""):
+    headers = {"Content-type": "application/json"}
+    url = f"{fqdn}{endpoint}" + (f"/{ticket_id}")
+    try:
+        response = requests.request(method, url, data=json.dumps(data), headers=headers, auth=(api_user, api_secret))
```

Comment:
Bin mir unsicher, ich glaube es ist besser statt request, das hier zu nutzen:

```
from ansible.module_utils.urls import fetch_url
```

Da macht dann die Libary schon einen Teil des Errror-Handlings. Ist aber auch nicht super schlimm, wenn du requests nimmst. Aber schau bitte mal.

Reply:
Most ansible modules I've come across are using the respective client libraries (e.g. pynetbox), which most likely use requests themselfes. I don't see an issue with using requests, it's more flexible than `fetch_url`